src/compiler/checker.ts

```diff
@@ -10670,7 +10671,7 @@ namespace ts {
                 const constraint = getConstraintOfTypeParameter(context.signature.typeParameters[index]);
                 if (constraint) {
                     const instantiatedConstraint = instantiateType(constraint, context);
-                    if (!isTypeAssignableTo(inferredType, getTypeWithThisArgument(instantiatedConstraint, inferredType))) {
+                    if (!context.compareTypes(inferredType, getTypeWithThisArgument(instantiatedConstraint, inferredType))) {

```

Comment:
is there anywhere else this could be used in place of a hard-coded call? I looked around and didn't see any, but I wonder if you noticed any while making this fix.